Typescript Errors when compiling

I'm getting Typescript errors when trying to build it (on both ```npm start``` or ```npm run build```.  I'm using Typescript 2.2.1, and haven't changed anything from the dependencies in your package.json - perhaps you can provide a yarn.lock?

I tried then targeting ES6 (changing both es5 references in tsconfig.json) and I get the following error...

targeting ES6:

```
ERROR in main.bba3d9a1cacf15ee9fad.js from UglifyJs
SyntaxError: Unexpected token: name (AuthModule) [main.bba3d9a1cacf15ee9fad.js:102,4]
```

targeting ES5 (intended target):

```
ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(47,36): error TS2304: Cannot find name 'Iterable'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(350,48): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(351,52): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(352,34): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(353,34): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(354,34): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(355,61): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(356,60): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(357,65): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(357,97): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(358,60): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(359,58): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(360,59): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(361,58): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(362,61): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(362,117): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(363,42): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(365,81): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(366,78): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(367,76): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(368,73): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(431,6): error TS2304: Cannot find name 'Symbol'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(464,59): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(465,59): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(467,48): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(468,69): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(472,40): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(474,48): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(494,55): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(495,45): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(511,46): error TS2304: Cannot find name 'Iterable'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(512,43): error TS2304: Cannot find name 'Iterable'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(627,18): error TS2304: Cannot find name 'PromiseConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(642,32): error TS2304: Cannot find name 'Iterable'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(646,39): error TS2304: Cannot find name 'Iterable'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(646,53): error TS2304: Cannot find name 'Iterator'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(1280,34): error TS2339: Property 'for' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(1284,41): error TS2339: Property 'hasInstance' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(1288,48): error TS2339: Property 'isConcatSpreadable' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(1292,38): error TS2339: Property 'iterator' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(1296,36): error TS2339: Property 'keyFor' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(1300,35): error TS2339: Property 'match' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(1304,37): error TS2339: Property 'replace' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(1308,36): error TS2339: Property 'search' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(1312,37): error TS2339: Property 'species' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(1316,35): error TS2339: Property 'split' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(1320,41): error TS2339: Property 'toPrimitive' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(1324,41): error TS2339: Property 'toStringTag' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(1328,41): error TS2339: Property 'unscopables' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(2059,34): error TS2339: Property 'for' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(2063,41): error TS2339: Property 'hasInstance' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(2067,48): error TS2339: Property 'isConcatSpreadable' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(2071,38): error TS2339: Property 'iterator' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(2075,36): error TS2339: Property 'keyFor' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(2079,35): error TS2339: Property 'match' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(2083,37): error TS2339: Property 'replace' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(2087,36): error TS2339: Property 'search' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(2091,37): error TS2339: Property 'species' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(2095,35): error TS2339: Property 'split' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(2099,41): error TS2339: Property 'toPrimitive' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(2103,41): error TS2339: Property 'toStringTag' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(2107,41): error TS2339: Property 'unscopables' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/rxjs/Observable.d.ts
(66,60): error TS2693: 'Promise' only refers to a type, but is being used as a value here.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@angular/core/src/facade/lang.d.ts
(12,17): error TS2693: 'Map' only refers to a type, but is being used as a value here.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@angular/core/src/facade/lang.d.ts
(13,17): error TS2693: 'Set' only refers to a type, but is being used as a value here.
12 03 2017 14:51:24.844:INFO [karma]: Karma v1.3.0 server started at http://localhost:9876/
12 03 2017 14:51:24.847:INFO [launcher]: Launching browser Chrome with unlimited concurrency
12 03 2017 14:51:24.906:INFO [launcher]: Starting browser Chrome
12 03 2017 14:51:26.141:INFO [Chrome 56.0.2924 (Mac OS X 10.12.3)]: Connected on socket /#RRuw3CmOMZ09-8zJAAAA with id 93742705
  auth/
    AuthService
      ✔ should be defined
      ✔ should subscribe to auth state changes
  tasks/
    Task
      ✔ should set title
      ✔ should set completed to false by default

Finished in 0.158 secs / 0.146 secs @ 14:51:27 GMT-0400 (EDT)

SUMMARY:
✔ 4 tests completed

> todo-angular2-firebase@0.8.0 build /Users/njonas/Code/todo-angular2-firebase
> cross-env NODE_ENV=production webpack --display-chunks --progress

 10% building modules 1/4 modules 3 active .../todo-angular2-firebase/src/vendor.tsts-loader: Using typescript@2.2.1 and /Users/njonas/Code/todo-angular2-firebase/tsconfig.json
Hash: 87c6ae3b1604d5928797
Version: webpack 2.1.0-beta.22
Time: 19411ms
                                          Asset       Size  Chunks             Chunk Names
                              browserconfig.xml  387 bytes          [emitted]
                 vendor.d8ebaac7f14ee065b6b0.js     769 kB       0  [emitted]  vendor
              polyfills.03ac86cc7d43b8f6ed7d.js      63 kB       2  [emitted]  polyfills
    styles.f5f666626e3a3317699512aecdacdbf0.css    1.34 kB       1  [emitted]  main
styles.f5f666626e3a3317699512aecdacdbf0.css.map  120 bytes       1  [emitted]  main
                     apple-touch-icon-57x57.png    1.02 kB          [emitted]
                   apple-touch-icon-144x144.png    2.76 kB          [emitted]
                   apple-touch-icon-114x114.png    1.79 kB          [emitted]
                     android-chrome-192x192.png      14 kB          [emitted]
                              favicon-16x16.png  812 bytes          [emitted]
                           apple-touch-icon.png    2.77 kB          [emitted]
                   main.040a6122430282ed5f55.js     349 kB       1  [emitted]  main
                     apple-touch-icon-72x72.png    1.19 kB          [emitted]
                              favicon-32x32.png    1.36 kB          [emitted]
                                  manifest.json  186 bytes          [emitted]
                                    favicon.ico    15.1 kB          [emitted]
                             mstile-150x150.png    5.05 kB          [emitted]
                             mstile-144x144.png    5.07 kB          [emitted]
                             mstile-310x310.png    20.1 kB          [emitted]
                             mstile-310x150.png    5.35 kB          [emitted]
                               mstile-70x70.png     3.4 kB          [emitted]
                          safari-pinned-tab.svg    2.49 kB          [emitted]
                                     index.html     1.9 kB          [emitted]
chunk    {0} vendor.d8ebaac7f14ee065b6b0.js (vendor) 1.98 MB {2} [initial] [rendered]
  [350] ./src async 160 bytes {0} [built]
     + 381 hidden modules
chunk    {1} main.040a6122430282ed5f55.js, styles.f5f666626e3a3317699512aecdacdbf0.css, styles.f5f666626e3a3317699512aecdacdbf0.css.map (main) 398 kB {0} [initial] [rendered]
  [599] multi main 28 bytes {1} [built]
     + 53 hidden modules
chunk    {2} polyfills.03ac86cc7d43b8f6ed7d.js (polyfills) 162 kB [entry] [rendered]
     + 164 hidden modules

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(47,36): error TS2304: Cannot find name 'Iterable'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(350,48): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(351,52): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(352,34): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(353,34): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(354,34): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(355,61): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(356,60): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(357,65): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(357,97): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(358,60): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(359,58): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(360,59): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(361,58): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(362,61): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(362,117): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(363,42): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(365,81): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(366,78): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(367,76): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(368,73): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(431,6): error TS2304: Cannot find name 'Symbol'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(464,59): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(465,59): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(467,48): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(468,69): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(472,40): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(474,48): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(494,55): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(495,45): error TS2304: Cannot find name 'PropertyKey'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(511,46): error TS2304: Cannot find name 'Iterable'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(512,43): error TS2304: Cannot find name 'Iterable'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(627,18): error TS2304: Cannot find name 'PromiseConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(642,32): error TS2304: Cannot find name 'Iterable'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(646,39): error TS2304: Cannot find name 'Iterable'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(646,53): error TS2304: Cannot find name 'Iterator'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(1280,34): error TS2339: Property 'for' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(1284,41): error TS2339: Property 'hasInstance' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(1288,48): error TS2339: Property 'isConcatSpreadable' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(1292,38): error TS2339: Property 'iterator' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(1296,36): error TS2339: Property 'keyFor' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(1300,35): error TS2339: Property 'match' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(1304,37): error TS2339: Property 'replace' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(1308,36): error TS2339: Property 'search' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(1312,37): error TS2339: Property 'species' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(1316,35): error TS2339: Property 'split' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(1320,41): error TS2339: Property 'toPrimitive' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(1324,41): error TS2339: Property 'toStringTag' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(1328,41): error TS2339: Property 'unscopables' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(2059,34): error TS2339: Property 'for' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(2063,41): error TS2339: Property 'hasInstance' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(2067,48): error TS2339: Property 'isConcatSpreadable' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(2071,38): error TS2339: Property 'iterator' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(2075,36): error TS2339: Property 'keyFor' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(2079,35): error TS2339: Property 'match' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(2083,37): error TS2339: Property 'replace' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(2087,36): error TS2339: Property 'search' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(2091,37): error TS2339: Property 'species' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(2095,35): error TS2339: Property 'split' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(2099,41): error TS2339: Property 'toPrimitive' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(2103,41): error TS2339: Property 'toStringTag' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@types/core-js/index.d.ts
(2107,41): error TS2339: Property 'unscopables' does not exist on type 'SymbolConstructor'.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/rxjs/Observable.d.ts
(66,60): error TS2693: 'Promise' only refers to a type, but is being used as a value here.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@angular/core/src/facade/lang.d.ts
(12,17): error TS2693: 'Map' only refers to a type, but is being used as a value here.

ERROR in /Users/njonas/Code/todo-angular2-firebase/node_modules/@angular/core/src/facade/lang.d.ts
(13,17): error TS2693: 'Set' only refers to a type, but is being used as a value here.
```

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -5,7 +5,7 @@
     "experimentalDecorators": true,
     "lib": [
       "dom",
-      "es5"
+      "es6"
     ],
     "module": "commonjs",
     "noEmitHelpers": true,

